File: Dataset/Leetcode/valid/19/115.java
 public ListNode XXX(ListNode head, int n) {
        if(head.next == null) return null;
        ListNode dumy = head;
        ListNode fast = head;
        for(int i=0;i<n;i++){
            fast = fast.next;
        }
        // 删除的是头节点
        if(fast == null) return head.next;
        while(fast.next != null){
            fast = fast.next;
            head = head.next;
        }
        head.next = head.next.next;
        return dumy;
    }

File: Dataset/Leetcode/train/2/115.py
class Solution:
    def XXX(self, l1: ListNode, l2: ListNode) -> ListNode:
        l1_list = []
        l2_list = []
        while l1 or l2:
            if l1:
                l1_list.append(l1.val)
                l1 = l1.next
            if l2:
                l2_list.append(l2.val)
                l2 = l2.next
        res = int(''.join([str(i) for i in l1_list[::-1]])) + int(''.join([str(i) for i in l2_list[::-1]]))
        res = [int(i) for i in str(res)][::-1]
        if res:
            temp = None
            for i in range(len(res)):
                temp = ListNode(res.pop(),temp)
            return temp
        else:
            return l1

